src/test/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerExecDecoratorWindowsTest.java

```diff
@@ -0,0 +1,227 @@
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Pattern;
+import org.apache.commons.io.output.TeeOutputStream;
+import org.apache.commons.lang.RandomStringUtils;
+import org.csanchez.jenkins.plugins.kubernetes.AllContainersRunningPodWatcher;
+import org.csanchez.jenkins.plugins.kubernetes.KubernetesClientProvider;
+import org.csanchez.jenkins.plugins.kubernetes.KubernetesCloud;
+import org.csanchez.jenkins.plugins.kubernetes.KubernetesSlave;
+import org.csanchez.jenkins.plugins.kubernetes.PodTemplate;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TestName;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.LoggerRule;
+import org.jvnet.hudson.test.recipes.WithTimeout;
+
+import static org.csanchez.jenkins.plugins.kubernetes.KubernetesTestUtil.assumeKubernetes;
+import static org.csanchez.jenkins.plugins.kubernetes.KubernetesTestUtil.assumeWindows;
+import static org.csanchez.jenkins.plugins.kubernetes.KubernetesTestUtil.deletePods;
+import static org.csanchez.jenkins.plugins.kubernetes.KubernetesTestUtil.getLabels;
+import static org.csanchez.jenkins.plugins.kubernetes.KubernetesTestUtil.setupCloud;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author Carlos Sanchez
+ */
+public class ContainerExecDecoratorWindowsTest {
+    @Rule
+    public ExpectedException exception = ExpectedException.none();
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    private KubernetesCloud cloud;
+    private static KubernetesClient client;
+    private static final Pattern PID_PATTERN = Pattern.compile("^((?:\\[\\d+\\] )?pid is \\d+)$", Pattern.MULTILINE);
+
+    private ContainerExecDecorator decorator;
+    private Pod pod;
+    private KubernetesSlave agent;
+
+    @Rule
+    public LoggerRule containerExecLogs = new LoggerRule()
+            .record(Logger.getLogger(ContainerExecDecorator.class.getName()), Level.ALL) //
+            .record(Logger.getLogger(KubernetesClientProvider.class.getName()), Level.ALL);
+
+    @Rule
+    public TestName name = new TestName();
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+        assumeKubernetes();
+        assumeWindows();
+    }
+
+    @Before
+    public void configureCloud() throws Exception {
+        cloud = setupCloud(this, name);
+        client = cloud.connect();
+        deletePods(client, getLabels(this, name), false);
+
+        String image = "mcr.microsoft.com/windows:10.0.17763.2686";
```

Comment:
Is this going to work on all node types?

Reply:
See https://github.com/jenkinsci/kubernetes-plugin/pull/1149/files#diff-560a4173f9aebfa545e3a529982bf114acda1a4642a7b4d7abc7f8b564ecf483R133, but I just committed a more selective node selector to clear any possible ambiguity.

Reply:
And if such a node does not exist in the pool will the test just fail? https://github.com/jenkinsci/kubernetes-plugin/blob/dc54b4000393fae433baf118250cd60451c64b0a/src/test/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesTestUtil.java#L160-L162 does not cover this.

Reply:
I guess this check can be improved to include the windows build as well.